classes/class-p4-search.php

```diff
@@ -43,45 +45,195 @@ public function __construct( $search_query, $templates = [ 'search.twig', 'archi
 				$this->context = $context;
 			} else {
 				$this->context = Timber::get_context();
+				$this->handle_submit( $this->context );
 
+				if ( $this->search_query ) {
+					$this->all_posts    = $this->get_posts();
+					$this->current_page = get_query_var( 'paged' ) ? get_query_var( 'paged' ) : 1;
+					$this->posts        = array_slice( $this->all_posts, ( $this->current_page - 1 ) * self::POSTS_PER_PAGE, self::POSTS_PER_PAGE );
+				}
+				$this->set_context( $this->context );
+			}
+		}
+
+		/**
+		 * Handle form submit.
+		 *
+		 * @param array $context Associative array with the data to be passed to the view.
+		 */
+		public function handle_submit( &$context ) {
+			if ( 'GET' === $_SERVER['REQUEST_METHOD'] ) {
```

Comment:
I think we should eliminate the use of superglobals in the class. This check (if request is GET) should be in an upper level (eg. search.php)